components/Carousel/Carousel.js

```diff
@@ -1,11 +1,55 @@
 class Carousel {
+    constructor(element) {
+        this.element = element;
+        this.leftButton = element.querySelector('.left-button');
+        this.rightButton = element.querySelector('.right-button');
+        this.images = this.element.querySelectorAll('img');
+        this.currentIndex = 0;
+        this.images[this.currentIndex].style.display = 'block';
 
+        this.leftButton.addEventListener('click', () => {
+            this.lastImg();
+        })
+
+        this.rightButton.addEventListener('click', () => {
+                this.nextImg();
+        })
+    }
+    
+
+    lastImg() {
+        if ((this.currentIndex - 1) < 0) {
+            this.currentIndex = this.images.length - 1;
+        } else {
+            this.currentIndex -= 1;
+        }
+        this.images.forEach(function (each) {
+            each.style.display = 'none';
+        })
+        this.images[this.currentIndex].style.display = 'block';
+    }
+
+
+    nextImg() {
+            if ((this.currentIndex + 1) > this.images.length - 1) {
+                this.currentIndex = 0;
+            } else {
+                this.currentIndex += 1;
+            }
+        this.images.forEach(function (each) {
+            each.style.display = 'none';
+        })
+        this.images[this.currentIndex].style.display = 'block';
+    }
 }
 
-let carousel = document.querySelector();
+
+let carousel = document.querySelectorAll('.carousel');
+
+carousel = Array.from(carousel).map(carousel => new Carousel(carousel));
```

Comment:
Amazing work on the carousel here 👌 The logic looks great and it works perfectly! 💯